spec/classes/init_spec.rb

```diff
@@ -794,6 +823,41 @@
           it { should contain_class('pam') }
         end
       end
+
+      context "with pam_d_sshd_template set to pam/sshd.custom.erb without specifying pam_sshd_[auth|account|password|session]_lines" do
+        let :facts do
+          { :osfamily => v[:osfamily],
+            :"#{v[:releasetype]}" => v[:release],
+            :lsbdistid => v[:lsbdistid],
+          }
+        end
+        let(:params) { { :pam_d_sshd_template => 'pam/sshd.custom.erb' } }
+
+        it "should fail" do
+            expect {
+              should contain_class('pam')
+            }.to raise_error(Puppet::Error, /pam_sshd_\[auth|account|password|session\]_lines required for template pam\/sshd.custom.erb/)
+        end
+      end
+
+      [ :pam_sshd_auth_lines, :pam_sshd_account_lines, :pam_sshd_password_lines, :pam_sshd_session_lines ].each do |param|
+        context "with #{param} specified and pam_d_sshd_template not specified on #{v[:osfamily]} with #{v[:releasetype]} #{v[:release]}" do
```

Comment:
Here [1] is a suggestion how to test that it fails when only one of the full param set is missing.

While verifing that the test is actully doing what's intended, I realized that even if the OR chain would fail the `validate_arry()` will fail later on too.

So maybe my commit is overdoing it. Anyway, was fun to code it ;)

[1] https://github.com/boandersson/puppet-module-pam/pull/1